File: modulo-diagnostico/src/main/java/org/tesis/modulodiagnostico/dtos/response/TopicPending.java
package org.tesis.modulodiagnostico.dtos.response;

import com.fasterxml.jackson.annotation.JsonProperty;

import java.math.BigDecimal;

public class TopicPending {
    @JsonProperty("topicId")
    private String topicId;
    @JsonProperty("topicName")
    private String topicName;
    @JsonProperty("domainLevel")
    private BigDecimal domainLevel;

    public String getTopicId() {
        return topicId;
    }

    public void setTopicId(String topicId) {
        this.topicId = topicId;
    }

    public String getTopicName() {
        return topicName;
    }

    public void setTopicName(String topicName) {
        this.topicName = topicName;
    }

    public BigDecimal getDomainLevel() {
        return domainLevel;
    }

    public void setDomainLevel(BigDecimal domainLevel) {
        this.domainLevel = domainLevel;
    }
}

File: motor_adaptabilidad/app/models/topic_pending.py
# app/models/topic_pending.py

class TopicPending:
    def __init__(self, topic_id: str, topic_name: str, domain_level: float):
        self.topic_id = topic_id
        self.topic_name = topic_name
        self.domain_level = domain_level

    def __repr__(self):
        return f"TopicPending({self.topic_id}, {self.topic_name}, {self.domain_level})"
